app/assets/javascripts/discourse/app/components/flash-message.gjs

```diff
@@ -0,0 +1,27 @@
+import Component from "@glimmer/component";
+import { concat } from "@ember/helper";
+import { action } from "@ember/object";
+import concatClass from "discourse/helpers/concat-class";
+
+const FLASH_TYPES = ["success", "error", "warning", "info"];
+
+export default class FlashMessage extends Component {
+  @action
+  validateFlashType(type) {
+    if (type && !FLASH_TYPES.includes(type)) {
+      throw `@type must be one of ${FLASH_TYPES.join(", ")}`;
+    }
+  }
+
+  <template>
+    {{this.validateFlashType @type}}
+    {{#if @flash}}
+      <div
+        class={{concatClass "alert" (if @type (concat "alert-" @type))}}
+        ...attributes
+      >
+        {{~@flash~}}
+      </div>
+    {{/if}}
```

Comment:
I think I would define a getter instead here:

```
get flashClass() {
  if (this.args.type && !FLASH_TYPES.includes(this.args.type)) {
    throw `@type must be one of ${FLASH_TYPES.join(", ")}`;
  }
  
  return this.args.type ? `alert-${this.args.type}` : null;
}
```

and then in template you can remove the function call and just do:

```
class={{concatClass "alert" this.flashClass}}
```